image/Dockerfile

```diff
@@ -72,4 +72,4 @@ RUN mkdir -p /project/dependencies/include && mkdir -p /project/dependencies/lib
 WORKDIR /project/build
 
 # Default build command will explicitly target wasm
-CMD /usr/local/Qt-wasm/bin/qt-cmake -DCMAKE_BUILD_TYPE=Release -G Ninja /project/source && ninja
+CMD /usr/local/Qt-wasm/bin/qt-cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE:-Release} -G Ninja /project/source && ninja
```

Comment:
Isn't the release vs. debug build mode already given by the Docker image, in the sense that it either contain a debug- or release-build of Qt?

I would then prefer that the application compiled here is always build in a consistent debug/release mode.

Reply:
There is no direct link between Qt and the app about debug and release configuration. (or in C/C++ in general when your link your app with external libraries)

You can have debug Qt and build your app in release and opposite, this could be even specified at the .cpp file compiled. If we want to keep it simple, you could easily create a Release version from a Debug version: just compile in debug and strip the debug symbols out from the binary.

Short version: you can debug your app, have the precise stack trace of the bug in your app without having Qt in debug mode. The compiler will show where it has debug symbol: the source code of your app.

(Remember that when you use the Qt installer, you download only the platform you want but can compile your app with all modes. If you want to debug Qt itself, you need to get Qt debug symbols from the installer.)

Reply:
> There is no direct link between Qt and the app about debug and release configuration. (or in C/C++ in general when your link your app with external libraries)

I know that's the case on Linux. However, on Windows you need to link to a debug-build of Qt when building an application in debug-mode, and a release-build of Qt when building an app in release-mode. The reason for that is that the memory layout (ABI) of certain C++ types differ between "debug" and "release" builds.

I would therefore like to follow the same pattern here, so that we stay consistent with existing practices on Windows.

Reply:
> Doing this totally disables the user to debug his application as he will need to create a debug-specific container.

Correct. That's why I'm already creating a separate "debug" container for debug builds. There's also a separate "debug" branch in this repo for testing debug builds against the debug container.